### ISSUE TITLE
When use amazon-s3, we got nothing if we pick a file by BB code!

### ISSUE THREAD

**Issue by JasonFun888** [LiveHelperChat/livehelperchat#1395]
I wonder that didn't anyone try the LiveHelperChat/amazon-s3 extension ?

When I use this extension, everything seems OK but you pick a file by BB code, such as [file=19_5304f561c06849121df0f4ba5cec39cd_img], and I got nothing if we pick a file by BB code！

I tried to find the reason, and fortunately I got it!

It's because of the mis-match way of generate and check the security_hash.
When we generate the BBCode, we use the full name of amazon s3,
[livehelperchat/lhc_web/lib/models/lhchat/erlhcoreclassmodelchatfile.php, Line 63](https://github.com/LiveHelperChat/livehelperchat/blob/bcb263542381544e035a78b7a6aa1c38de8b3efc/lhc_web/lib/models/lhchat/erlhcoreclassmodelchatfile.php#L63):
```PHP
public function __get($var){
   		switch ($var) {
   			...
   			case 'security_hash':
   					$this->security_hash = md5($this->name.'_'.$this->chat_id);
   				return $this->security_hash;
   			;
   			break;
```

but when we check the BBCode, we use always use original name,
[lhc_web/lib/core/lhbbcode/lhbbcode.php, Line 800](https://github.com/LiveHelperChat/livehelperchat/blob/a1527168909805be8a5e26fb1353946556e2a386/lhc_web/lib/core/lhbbcode/lhbbcode.php#L800):
```PHP
public static function _make_url_file($matches)
   {
  		if (isset($matches[1])){
   		    ...
   				if (is_object($file)) {
                    // AWS plugin changes file name, but we always use original name
                    $parts = explode('/', $file->name);
                    end($parts);
                    $name = end($parts);

                    // Check that user has permission to see the chat. Let say if user purposely types file bbcode
                    if ($hash == md5($name . '_' . $file->chat_id)) {
``` 

Dear @remdex , would you please check if this issue really exists and fix it or make them all work better?
BTW, I have opened a _pull request_([For AWS S3 plugin, we always use original name #1394](https://github.com/LiveHelperChat/livehelperchat/pull/1394)) , if it works it's my extraordinary pleasure to do something on the Live Helper Chat!
Thanks!
And thanks for your share!

**Comment by remdex** [comment]
Thank you! I have merged your pull request :)

**Comment by JasonFun888** [comment]
Bingo! My pleasure!

**Comment by JasonFun888** [comment]
> Thank you! I have merged your pull request :)

Bingo! My pleasure!

**Comment by JasonFun888** [comment]
@remdex Hi, dear master, 
I found another issue after last fix that I can NOT download a file from the file_url, cause there are several places need to check the $file->security_hash, and each of them should be updated.
It's up to the erLhcoreClassModelChatFile Class to make rule of the security_hash, every other just get the it for checking permission.

I have created three Pull requests on some three repos, 

- [Jason fun888 patch 1 fix the mismatched security hash #1399](https://github.com/LiveHelperChat/livehelperchat/pull/1399)
- [For AWS S3 plugin, we always use original name #16](https://github.com/LiveHelperChat/fbmessenger/pull/16)
- [For AWS S3 plugin, we always use original name #4](https://github.com/LiveHelperChat/livehelperchat-extensions/pull/4)

Please check it, thanks!
